homeassistant/components/switchbot_cloud/sensor.py

```diff
@@ -90,15 +70,11 @@ async def async_setup_entry(
     async_add_entities(
         SwitchBotCloudSensor(data.api, device, coordinator, description)
         for device, coordinator in data.devices.sensors
-        for description in METER_PRO_CO2_SENSOR_DESCRIPTIONS
-        if device.device_type == "MeterPro(CO2)"
-    )
-
-    async_add_entities(
-        SwitchBotCloudSensor(data.api, device, coordinator, description)
-        for device, coordinator in data.devices.sensors
-        for description in METER_PLUS_SENSOR_DESCRIPTIONS
-        if device.device_type != "MeterPro(CO2)"
+        for description in (
+            METER_PRO_CO2_SENSOR_DESCRIPTIONS
+            if device.device_type == "MeterPro(CO2)"
+            else METER_PLUS_SENSOR_DESCRIPTIONS
+        )
```

Comment:
Let's assign this to a variable first before, and then iterate over the variable, I think that is a lot cleaner :)

Reply:
Sorry, I'm sort of a newbie to python list comprehension. Are you saying before the list comprehension? Don't we need to iterate over devices to know the device type?

Reply:
```py
descriptions = METER_PRO_CO2_SENSOR_DESCRIPTIONS
            if device.device_type == "MeterPro(CO2)"
            else METER_PLUS_SENSOR_DESCRIPTIONS
```
and then have
```py
for description in descriptions
```

Reply:
Sorry, again a newbie here.

If we broke it out like that before the list comprehension, wouldn't `device.device_type` be undefined? We need to have the `if device.device_type == "MeterPro(CO2)"` condition in the list comprehension as it iterates over `data.devices.sensors`

Reply:
Oh you are correct, I missed that the device was part of the for loop